packages/allure-cypress/src/index.ts

```diff
@@ -0,0 +1,188 @@
+  cy.now("allureReporterMessage", {
+    type: MessageType.METADATA,
+    payload: {
+      parameter: [{ name, value, ...options }],
+    },
+  });
+};
+export const description = (markdown: string) => {
+  cy.now("allureReporterMessage", {
+    type: MessageType.METADATA,
+    payload: {
+      description: markdown,
+    },
+  });
+};
+export const descriptionHtml = (html: string) => {
+  cy.now("allureReporterMessage", {
+    type: MessageType.METADATA,
+    payload: {
+      descriptionHtml: html,
+    },
+  });
+};
+export const testCaseId = (value: string) => {
+  cy.now("allureReporterMessage", {
+    type: MessageType.METADATA,
+    payload: {
+      testCaseId: value,
+    },
+  });
+};
+export const historyId = (value: string) => {
+  cy.now("allureReporterMessage", {
+    type: MessageType.METADATA,
+    payload: {
+      historyId: value,
+    },
+  });
+};
+export const allureId = (value: string) => {
+  cy.now("allureReporterMessage", {
+    type: MessageType.METADATA,
+    payload: {
+      labels: [{ name: LabelName.ALLURE_ID, value }],
+    },
+  });
+};
+export const displayName = (name: string) => {
+  cy.now("allureReporterMessage", {
+    type: MessageType.METADATA,
+    payload: {
+      displayName: name,
+    },
+  });
+};
+export const issue = (url: string, name?: string) => {
+  link(url, name, LinkType.ISSUE);
+};
+export const tms = (url: string, name?: string) => {
+  link(url, name, LinkType.TMS);
+};
+export const epic = (name: string) => {
+  label(LabelName.EPIC, name);
+};
+export const feature = (name: string) => {
+  label(LabelName.FEATURE, name);
+};
+export const story = (name: string) => {
+  label(LabelName.STORY, name);
+};
+export const suite = (name: string) => {
+  label(LabelName.SUITE, name);
+};
+export const parentSuite = (name: string) => {
+  label(LabelName.PARENT_SUITE, name);
+};
+export const subSuite = (name: string) => {
+  label(LabelName.SUB_SUITE, name);
+};
+export const owner = (name: string) => {
+  label(LabelName.OWNER, name);
+};
+export const severity = (name: string) => {
+  label(LabelName.SEVERITY, name);
+};
+export const layer = (name: string) => {
+  label(LabelName.LAYER, name);
+};
+export const tag = (name: string) => {
+  label(LabelName.TAG, name);
+};
+export const attachment = (name: string, content: unknown, type: string, encoding: string = "utf8") => {
+  const objectAttachment = typeof content === "object";
+
+  // non-object attachment is a string and fully controllable by user
+  if (!objectAttachment) {
+    cy.now("allureReporterMessage", {
+      type: MessageType.METADATA,
+      payload: {
+        attachments: [
+          {
+            content: content as string,
+            encoding: encoding as BufferEncoding,
+            name,
+            type,
+          },
+        ],
+      },
+    });
+    return;
+  }
+
+  let attachmentContent: string;
+
+  switch ((content as CypressWrappedAttachment).type) {
+    case "Buffer":
```

Comment:
Should we support the following scenario?

```ts
it("test", () => {
    cy.readFile("img/allure.png", null).then((c) => {
        allure.attachment("logo", c, "image/png");
    });
});
```

An example would be downloading a file from the website under test via [cypress-downloadfile](https://www.npmjs.com/package/cypress-downloadfile). Seems legit, but currently throws:

```
Error:    CypressError: `cy.task('allureReportTest')` failed with the following error:

> The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received undefined

Because this error occurred during a `after each` hook we are skipping all of the remaining tests.
```

> [!NOTE]
> I've passed `null` to `readFile` to read it as `Buffer` (see [here](https://docs.cypress.io/api/commands/readfile)). But it appears that its `.data` field is undefined.

Reply:
Let's add the logic in next iterations because the logic is provided by third-party plugin, not by the core functionality